config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,13 @@
+LIMIT_LOGIN_ATTEMPTS_PER_IP = 16
+
+Rack::Attack.throttle("limit login attempts per IP", limit: LIMIT_LOGIN_ATTEMPTS_PER_IP, period: 1.hour) do |request|
+  is_welcome_page = request.path == "/" && request.params.dig("user", "email").present?
+  is_login_page = request.path == "/login"
+  if (is_welcome_page || is_login_page) && request.post?
+    (request.env["action_dispatch.remote_ip"] || request.ip).to_s
```

Comment:
just out of curiosity - I'm not clear on the difference between the two potential sources of IP and why action_dispatch's remote_ip might be nil

Reply:
In this block we've got a `Rack::Request` not an `ActionDispatch::Request` so the `remote_ip` method is unavailable.  I copied the definition [from rails](https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/http/request.rb#L293).

...but having thought about it, I think `request.ip` will always be a PaaS loadbalancer (or maybe a Cloudfront node) in production / staging, which we never want to throttle.  I'll remove it.